carbon-language/carbon-lang
# Documentation website

Closes #1492. Adds a website that is auto-generated from the markdown files in the repo, using [Docusaurus](https://docusaurus.io/docs). This will make it easier to browse the docs, by adding a navigation sidebar as well as a table-of-contents sidebar, mobile support, and other things. The site can be autodeployed to https://carbon-language.github.io/ (or to a custom domain using a CNAME file), and is currently configured accordingly (deployment instructions in [website/README.md](https://github.com/carbon-language/carbon-lang/pull/1526/files#diff-0eb684c3ea7be1b3de6b0b158c7059da05dff87877c4c40d32ea3a8eaef7a75eR43-R53)).

- The front page is generated from the repository root readme.
- The top navigation items Design, Project, Guides, and Spec corresponds to the respective directories under `docs/`.
- Footer has links to GitHub repo, Discord, and Code of Conduct.

Preview: https://emlai.github.io/

To do:
- [ ] Render [mermaid diagrams](https://github.com/carbon-language/carbon-lang/pull/1526#issuecomment-1227782614). 
- [ ] Before or after merging, for someone with admin access to the organization: Create carbon-language.github.io repo, enable GitHub Pages for it, and configure deploy keys ([instructions](https://github.com/carbon-language/carbon-lang/pull/1526/files#diff-0eb684c3ea7be1b3de6b0b158c7059da05dff87877c4c40d32ea3a8eaef7a75eR43-R53)). You can ping me (emlai#9471) on Discord if you need help.

Follow-up improvements (non-blocking tasks that are not necessary for merging this PR):
- [ ] Mention in docs that they get auto-deployed to the website when a change is pushed to trunk.
- [ ] Add site-wide search such as [Algolia](https://docsearch.algolia.com/apply/) which will scrape the site and host the search index in their own database. Requires that the site is deployed first before setting up.
- [ ] Fix broken links and enable Docusaurus link validation ([src](https://github.com/carbon-language/carbon-lang/pull/1526/files#diff-bc4913d026ae51bef2bc8b7a1c040fdf52b999d99e6fa40d87b379f98a60e0eaR16)), blocked by Docusaurus issue: https://github.com/facebook/docusaurus/issues/7829
- [ ] Fix description meta tags, currently `<!--`: `<meta data-rh="true" name="description" content="<!--">`

## How to test

Go to https://emlai.github.io/ and check that links, navigation, etc. works, and everything looks ok.
